app/services/RoutingService.scala

```diff
@@ -44,20 +46,33 @@ class RoutingService @Inject() (appConfig: AppConfig, messageConnector: MessageC
           officeOfDeparture(rootXml)
         }
 
-        officeEither.map {
+        officeEither.flatMap {
           office =>
-            if(office.value.startsWith("XI")) {
-              logger.info("routing to NI")
-              messageConnector.post(xml.toString(), appConfig.eisniUrl, appConfig.eisniBearerToken)
-            }
-            else {
-              logger.info("routing to GB")
-              messageConnector.post(xml.toString(), appConfig.eisgbUrl, appConfig.eisgbBearerToken)
+            getEisDetails(office, request.channel).map {
+              details => messageConnector.post(xml.toString(), details.url, details.token)
             }
         }
     }
   }
 
+  private case class EisDetails(url: String, token: String, routingMessage: String)
+
+  private def getRoutingOption(office: Office, channelType: ChannelType) =
+    (office.value.startsWith("XI"), channelType) match {
```

Comment:
Can we use the type system to help here, rather than matching on magic strings? E.g. can Office have a method that tells you whether it is GB, XI or something else?

Reply:
👍